ocaml-lsp-server/src/code_actions/action_remove_type_annotation.ml

```diff
@@ -0,0 +1,77 @@
+open Import
+
+let action_kind = "remove type annotation"
+
+let check_typeable_context pipeline pos_start =
+  let pos_start = Mpipeline.get_lexing_pos pipeline pos_start in
+  let typer = Mpipeline.typer_result pipeline in
+  let browse = Mbrowse.of_typedtree (Mtyper.get_typedtree typer) in
+  let is_exp_constrained = function
+    | Typedtree.Texp_constraint _, loc, _ -> Some loc
+    | Typedtree.Texp_coerce (Some { ctyp_loc; _ }, _), _, _ -> Some ctyp_loc
+    | _ -> None
+  in
+  let is_pat_constrained = function
+    | Typedtree.Tpat_constraint _, loc, _ -> Some loc
+    | _ -> None
+  in
+  let is_valid loc p extras =
+    match extras |> List.rev |> List.find_map ~f:p with
```

Comment:
Could you leave a comment as to why we're searching from the end?

Reply:
Sure. I hope it's clear enough. :)